### ISSUE TITLE
The triplet loss converges to the margin

### ISSUE THREAD

**Issue by viyjy** [wanggrun/triplet#1]
Hi, I have tried to replace infoNCE loss by triplet loss (using hardest negative sample), but the triplet loss simply converges to the margin, resulting in collapsed model where all examples have the same embedding. Any suggestion? 

BTW, would you please share the loss function you used in your paper? Thanks.

**Comment by wanggrun** [comment]
Thanks so much for your interest in our work! As our paper suggests, the hardest negative sample has several problems, and using smoothed-rank-k triplet loss is our main contribution. Please use the smoothed-rank-k triplet loss.

Our full code will be here soon.

**Comment by viyjy** [comment]
Thanks for your reply. Another question is: what is median_triplet_head.py? Does it correspond to any number in the paper?

**Comment by wanggrun** [comment]
@viyjy median_triplet_head.py is 73.6 in the paper. As is mentioned by the paper, median triplet loss is not as good as smoothed-rank-k triplet loss, especially when training longer.

**Comment by viyjy** [comment]
Thanks very much.
